packages/fxa-shared/sentry/index.ts

```diff
@@ -1,3 +1,68 @@
 export * from './tag';
 export * from './config-builder';
 export * from './models/SentryConfigOpts';
+import { Hub, Severity } from '@sentry/types';
+
+/** Minimal type containing validation error fields that are used by routines. */
+export type ValidationError = {
```

Comment:
Should these extend Error? I noticed some of these `*Error` objects are types (here), some are classes, and sometimes they extend `Error`. It's not clear when its what, or why.

Reply:
Hmm, these objects are not being 'thrown' so not sure if it matters. They were just adding a bit of type safety, so a message could be built up. That being said, the underlying objects being sent in are joi validations errors. I didn't realize fxa-shared had an existing joi dependency already, so I can just use that instead.

Reply:
> 

Thanks for the review. I know this is a rather large change set. I am working on addressing the feedback. Here's the main things I will address in a second push:

1. Hoist the entirety of UserManager.queryCurrentSubscriptions to fxa-shared, which will ensnare other things.
2. Fix a couple fields that got accidentally marked as public.
3. See if I can work around passing the sentry instance to reportValidationError and address ValidationError type there.

Reply:
Alright, this has been moved to fxa-shared/sentry/report-validation-error.ts, and the now references the joi type for ValidationError.